openassessment/xblock/static/js/src/lms/oa_response.js

```diff
@@ -836,14 +825,13 @@ OpenAssessment.ResponseView.prototype = {
 
         sel.find('.file__upload').prop('disabled', true);
 
-        promise = view.removeUploadedFiles();
-        promise = promise.then(function() {
-            return view.saveFilesDescriptions();
-        });
+        promise = view.saveFilesDescriptions();
 
+        view.fileCountBeforeUpload = view.getSavedFileCount();
```

Comment:
well, so here I believe you do need the total count of all files, not just the current count of non-deleted files

Reply:
I disagree. Do you want to chat?

Reply:
since we limit the user based on the number of "real"/non deleted files - it doesn't make sense to take deleted files into account.

Reply:
Per our conversation, changes made.